src/main/java/nerf/task/TaskList.java

```diff
@@ -0,0 +1,246 @@
+     */
+    public TaskList(List<String> taskList) {
+        this.listings = new ArrayList<>();
+        for (String taskLine: taskList) {
+            String[] task = taskLine.split("\\|");
+            switch (task[0].trim()) {
+            case "T" -> listings.add(new ToDos(task[2].trim(), task[1].trim().equals("1")));
+            case "D" -> {
+                LocalDate dueDate = Parser.parseStringToDate(task[3].trim());
+                listings.add(new Deadlines(task[2].trim(), task[1].trim().equals("1"), dueDate));
+            }
+            case "E" -> {
+                LocalDate fromDate = Parser.parseStringToDate(task[3].trim());
+                LocalDate toDate = Parser.parseStringToDate(task[4].trim());
+                listings.add(new Events(task[2].trim(), task[1].trim().equals("1"), fromDate, toDate));
+            }
+            default -> System.out.println("Save file seems to be corrupted.");
+            }
+        }
+    }
+
+    /**
+     * Print current list of tasks.
+     * 
+     * @return list of tasks.
+     */
+    public String printList() {
+        String res = "Here are the task(s) in your list:\n";
+        for (int i = 0; i < listings.size(); i++) {
+            res += String.format("%d. %s", i + 1, listings.get(i)) + "\n";
+        }
+        return res;
+    }
+
+    /**
+     * Set task completion on specified task.
+     * 
+     * @param input specific index of task.
+     * @return Mark task confirmation.
+     */
+    public String markTask(String input) {
+        String number = input.substring(4).trim();
+        try {
+            int num = Integer.parseInt(number);
+            listings.get(num - 1).setDone();
+            return  """
+                    Wow! Congrats on finishing another task!
+                    I've marked this task as done:\n
+                    """ + listings.get(num - 1);
+        } catch (NumberFormatException e) {
+            return "Oops! That does not seem like a number.";
+        } catch (IndexOutOfBoundsException e) {
+            return "Oops! Please specific a number within the list.";
+        }
+    }
+
+    /**
+     * Unset task completion on specified task.
+     * 
+     * @param input specific index of task.
+     * @return unmark task confirmation.
+     */
+    public String unmarkTask(String input) {
+        String number = input.substring(6).trim();
+        try {
+            int num = Integer.parseInt(number);
+            listings.get(num - 1).setUndone();
+            return  """
+                    Ok noted!
+                    I've marked this task as undone:\n
+                    """ + listings.get(num - 1);
+        } catch (NumberFormatException e) {
+            return "Oops! That does not seem like a number.";
+        } catch (IndexOutOfBoundsException e) {
+            return "Oops! Please specific a number within the list.";
+        }
+    }
+
+    /**
+     * Filter and find task based on keyword.
```

Comment:
Coding standard violation: I believe `Filters` should be used instead of `Filter` for JavaDocs